nixos/modules/services/x11/xserver.nix

```diff
@@ -313,12 +335,27 @@ in
 
       xrandrHeads = mkOption {
         default = [];
-        example = [ "HDMI-0" "DVI-0" ];
-        type = with types; listOf string;
+        example = [ "HDMI-0" { output = "DVI-0"; primary = true; monitorConfig = ""; } ];
+        type = with types; listOf (either
+          str
+          (submodule {
+            options.output = str;
+            options.primary = bool;
+            options.monitorConfig = lines;
```

Comment:
Yep, I identified that when leaving out `monitorConfig` it fails. That doesn't change just because I add `mkOption` with a `default` parameter. I did that before, and it still failed. So I was wondering what functional difference is adding `mkOption`. Perhaps it is a feature that hasn't been implemented yet?